aiplatform/snippets/embeddings_preview.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+// [START generativeaionvertexai_sdk_embedding]
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	aiplatform "cloud.google.com/go/aiplatform/apiv1"
+	aiplatformpb "cloud.google.com/go/aiplatform/apiv1/aiplatformpb"
+
+	"google.golang.org/api/option"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+func If[T any](cond bool, vtrue, vfalse T) T {
+	if cond {
+		return vtrue
+	}
+	return vfalse
+}
+
+func embedTextsPreview(
+	apiEndpoint, project, model string, texts []string,
```

Comment:
I wonder if the API endpoint is the best thing to expose as a parameter here, given the user will always use the production `{region}-aiplatform.com` endpoint--could this be `location` instead?

Reply:
Let's handle this as a feature request, because other samples (nodejs, java, and python) are accepting api_endpoint (not location).